### PR TITLE
[resotolib][fix] Make config cache-friendly

### PR DESCRIPTION
**Pull request by meln1k**
# Description

# To-Dos

- [x] Add test coverage for new or updated functionality
- [x] Lint and test with `tox`
- [ ] Document new or updated functionality (someengineering/resoto.com#XXXX)

# Issues Fixed

- Fixes #XXXX

# Code of Conduct

By submitting this pull request, I agree to follow the [code of conduct](https://resoto.com/code-of-conduct).

### CODE REVIEW

File: resotolib/resotolib/config.py [hunk 1]
```diff
@@ -264,6 +264,16 @@ def on_config_event(self, message: Dict[str, Any]) -> None:
             except Exception:
                 log.exception("Failed to reload config")
 
+    # the __hash__ and the __eq__ below is a workaround to make sure the outdated config is not cached by
+    # a lru_cache decoartor after the config performed a self-update. It serves no other purpose.
+    def __hash__(self) -> int:
+        return self.running_config.revision.__hash__()
```
Comment (aquamatthias) [inline resotolib/resotolib/config.py:270]:
```suggestion
        return hash(self.running_config.revision)
```

File: resotolib/resotolib/config.py [hunk 2]
```diff
@@ -264,6 +264,16 @@ def on_config_event(self, message: Dict[str, Any]) -> None:
...
+
+    def __eq__(self, other: Any) -> bool:
+        if isinstance(other, Config):
+            return self.running_config.__dict__ == other.running_config.__dict__
```
[the remark this answers is not in the document]
Comment (meln1k) [reply resotolib/resotolib/config.py:274]: Do you mean without the whole dict?

[the remark this answers is not in the document]
Comment (aquamatthias) [reply resotolib/resotolib/config.py:274]: I guess it is fine.

**Review by aquamatthias — APPROVED** [review]
Thanks

**Pull request merged**